### ISSUE TITLE
add hlint to CI builders

### ISSUE THREAD

**Issue by agentm** [agentm/project-m36#127]
n/t

**Comment by 3noch** [comment]
Do you already have an hlint configuration that you use? The latest hlint allows a yaml file configuration which seems highly preferable for CI.

**Comment by agentm** [comment]
I haven't used hlint on this project at all. If you have a config to share, that would be much appreciated.

**Comment by agentm** [comment]
Ugh. hlint crashes in cpphs:

```
hlint: #error PROJECTM36_VERSION is not defined
in src/bin/TutorialD/tutd.hs  at line 75 col 1
CallStack (from HasCallStack):
  error, called at ./Language/Preprocessor/Cpphs/CppIfdef.hs:113:21 in cpphs-1.20.8-JLeLxWRxjmm1BBYuZIZezR:Language.Preprocessor.Cpphs.CppIfdef
```

**Comment by 3noch** [comment]
I presume you tried the latest version?

**Comment by agentm** [comment]
Yes. I was able to resolve this by adding a command line flag to define the CPP macro.

```
hlint src --cpp-define PROJECTM36_VERSION=0.1
```

I also opened [an hlint ticket](https://github.com/ndmitchell/hlint/issues/371).

I currently see 380 hints for ```src```, so I have some work ahead to fix or squelch them.

**Comment by agentm** [comment]
Merged to master in 51915498c2dda1c0800368b6b265fbd49745cabd.
